### PR TITLE
CLOUDP-57632: Implementing Alerts resource

### PR DESCRIPTION
**Pull request by andreaangiolillo**
[CLOUDP-57632](https://jira.mongodb.org/browse/CLOUDP-57632): 
Implement the Atlas API resource for alerts, https://docs.atlas.mongodb.com/reference/api/alerts/

### CODE REVIEW

File: mongodbatlas/atlas_alerts.go [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+
+// Alert represents MongoDB Alert.
+type Alert struct {
+	ID                     string           `json:"id,omitempty"`                     // Unique identifier.
+	GroupID                string           `json:"groupId,omitempty"`                // Unique identifier of the project that owns this alert configuration.
+	AlertConfigID          string           `json:"alertConfigId,omitempty"`          // ID of the alert configuration that triggered this alert.
+	EventTypeName          string           `json:"eventTypeName,omitempty"`          // The type of event that will trigger an alert.
+	Created                string           `json:"created,omitempty"`                // Timestamp in ISO 8601 date and time format in UTC when this alert was opened.
+	Updated                string           `json:"updated,omitempty"`                // Timestamp in ISO 8601 date and time format in UTC when this alert was last updated.
+	Enabled                *bool            `json:"enabled,omitempty"`                // If omitted, the configuration is disabled.
```
Comment (gssbzn) [inline mongodbatlas/atlas_alerts.go]: I'd rather not make this a pointer and not omit it as the behaviour would be the same but I don't have a strong opinion against this

File: mongodbatlas/atlas_alerts.go [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+	CurrentValue           *CurrentValue    `json:"currentValue,omitempty"`           // CurrentValue represents current value of the metric that triggered the alert. Only present for alerts of type HOST_METRIC.
+	ReplicaSetName         string           `json:"replicaSetName,omitempty"`         // Name of the replica set. Only present for alerts of type HOST, HOST_METRIC, BACKUP, and REPLICA_SET.
+	ClusterName            string           `json:"clusterName,omitempty"`            // The name the cluster to which this alert applies. Only present for alerts of type BACKUP, REPLICA_SET, and CLUSTER.
+	Matchers               []Matcher        `json:"matchers,omitempty"`               // You can filter using the matchers array only when the EventTypeName specifies an event for a host, replica set, or sharded cluster.
+	MetricThreshold        *MetricThreshold `json:"metricThreshold,omitempty"`        // MetricThreshold  causes an alert to be triggered.
+	Notifications          []Notification   `json:"notifications,omitempty"`          // Notifications are sending when an alert condition is detected.
+}
+
+// AlertsRequest contains the request Body Parameters
+type AlertsRequest struct {
```
Comment (gssbzn) [inline mongodbatlas/atlas_alerts.go]: [nit] can we change this?
```suggestion
type AcknowledgeRequest struct {
```

File: mongodbatlas/atlas_alerts.go (hunk 1 above)
Comment (andreaangiolillo) [reply mongodbatlas/atlas_alerts.go]: The alert_configurations is using a pointer for Enabled. If we decided to change this pointer, I think we should also change the one in alert_configurations.

Comment (gssbzn) [reply mongodbatlas/atlas_alerts.go]: Ok let's leave it as it is for the sake of continuity

File: mongodbatlas/atlas_alerts_test.go [hunk 3]
```diff
@@ -0,0 +1,301 @@
...
+
+import (
+	"fmt"
+	"github.com/go-test/deep"
+	"github.com/mwielbut/pointy"
+	"net/http"
+	"testing"
+)
+
+func TestAlert_GetAnAlert(t *testing.T) {
```
Comment (gssbzn) [inline mongodbatlas/atlas_alerts_test.go]: Your tests still reference the old name, let's keep them in sync please

**Comment by gssbzn** [comment]
@themantissa @marinsalinas we updated this with latest master and should be ready for final review

**Pull request merged**